src/Integration/Helpers/ToolWIndowService.cs

```diff
@@ -42,7 +42,10 @@ public ToolWindowService([Import(typeof(SVsServiceProvider))] IServiceProvider s
 
         public void Show(Guid toolWindowId)
         {
-            var hr = shell.FindToolWindow(0, toolWindowId, out var windowFrame);
+            // We want VS to ask the package to create the tool window if it doesn't already exist
+            const uint flags = (uint)__VSFINDTOOLWIN.FTW_fForceCreate;
+
+            var hr = shell.FindToolWindow(flags, toolWindowId, out var windowFrame);
```

Comment:
Previously the service was passing 0 in the call to VS, which the documentation doesn't list as a valid value. I thought it was odd at the time, but it's what the previous code was doing so I didn't change it.

However, it does make a difference now, and "force create" is the behaviour we want.

It works like this: well-known tool window X may or may not exist. The code calling the shell doesn't know whether the window exists yet or not, and shouldn't  have to care. The calling code also doesn't know which package implements the tool window, and again it shouldn't have to.

Instead, the calling code calls shell with "force create" set. VS will check whether the window already exists. If it does, VS will just show it. If it doesn't, it will work out which package implements the tool window and ask it to create it. Then VS will show it.

I've manually tested it and it works the way we want. On the first call, VS will ask the package to create the window. On subsequent calls it doesn't.